lib/libc/Kconfig

```diff
@@ -40,7 +40,8 @@ endchoice # LIBC_IMPLEMENTATION
 
 if NEWLIB_LIBC
 
-if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb"
+if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb" || \
+  ("$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr" && "${SDK_MAJOR_MINOR}" = "0.11")
 config NEWLIB_LIBC_NANO
 	bool "Build with newlib-nano C library"
 	default y
```

Comment:
Why is it needed to enable this by default if it is a tool-chain dependent option? I know this is not part of your change, but I think it is a bad a idea to have this filtering based on SDK being used, what if I have another customer toolchain that supports that?

I think we should remove the conditional and disable this by default and let users enable the Kconfig if their toolchain support newlib-nano and not do that based on toolchain name as defined by zephyr.

Reply:
IMHO, toolchains should be expressing what they support with custom Kconfig sources.
Just like SoC's, arch's, and board's are already doing.

So there should exist a HAS_NEWLIB_LIBC_NANO option that is sourced through a toolchain-specific Kconfig file.

Then here we could default to enabling NANO when it is supported.

Disabling NEWLIB_LIBC_NANO by default would increase code size.